corehq/apps/app_manager/views/utils.py

```diff
@@ -350,6 +351,12 @@ def update_linked_app(app, master_app_id_or_build, user_id):
     ):
         old_multimedia_ids = set([media_info.multimedia_id for path, media_info in app.multimedia_map.items()])
         report_map = get_static_report_mapping(master_build.domain, app['domain'])
+        if not app.domain_link.is_remote:
+            report_map.update({
+                c.report_meta.master_id: c._id
+                for c in get_report_configs_for_domain(app.domain)
+                if c.report_meta.master_id
```

Comment:
@orangejenny should https://github.com/dimagi/commcare-hq/pull/29257/files#diff-7630fbc8238aa62c920becfd61d929e79daa9458b678026cb250c1da69fb2f53R353 be added here as well. I am thinking ReportConfiguration could have both linked and normal domains, and in the case of former master_id would be `None`

Reply:
Ha, I thought about this on my first review pass and decided it was too much of a nitpick to comment on. Without the check, there'll just be a `None` key in the report map that'll never get read, so it'll work fine, but I do think it's a little weird.

Reply:
better to have none instead of None :)